### ISSUE TITLE
Extra comma

### ISSUE THREAD

**Issue by thomasballinger** [elm-lang/error-message-catalog#168]
Elm-format and the Elm community's preference for commas at the beginning of lines sometimes trips me up because I'm in the habit of putting commas at the end of lines. I've been momentarily confused by something like this a few times:

```
type alias Drag =
    { start : Position,
    , current : Position
    , clipId : Integer
    }

-- SYNTAX PROBLEM ----------------------------------------------------- Main.elm

I ran into something unexpected when parsing your code!

61|     , current : Position,
        ^
I am looking for one of the following things:

    a lower case name
    whitespace
```

I wonder if this is command enough and specific enough to merit a suggestion for what the problem might be. The syntax highlighting I use has Types and commas the same color, so I often miss a trailing comma I put somewhere by habit.

**Comment by Rolograaf** [comment]
Maybe this is different or maybe the same. As European I am used that Floats are noted with a comma instead a point, in programming usually the US way is used. However the error message is not very useful?

```
I ran into something unexpected when parsing your code!

3|   ceiling 9,32
              ^
I am looking for one of the following things:

    an expression
    an infix operator like (+)
    end of input
    whitespace
```

**Comment by evancz** [comment]
@thomasballinger, the new error is like this:

![screen shot 2016-12-15 at 8 00 56 pm](https://cloud.githubusercontent.com/assets/1658058/21249078/3a47a536-c301-11e6-828a-cd20bd0bbd7e.png)

So now both commas show up at least. If you think it can still be better, you can reopen with a suggestion of how it should look. Either way, thanks for the SSCCE!

@Rolograaf, that is a different thing. Prefer to open new issues. Having users try to avoid duplicate issues leads to bad outcomes. (1) It is valuable to hear the same thing from different perspectives. (2) Without lots of knowledge, it is hard to tell if something *is* a duplicate at all. (3) If there are many duplicates, they can be combined (by a maintainer) into a meta issue that is much clearer.

I'll try to do something nice for that case though. Will update soon!

**Comment by evancz** [comment]
Ah, detecting `3,14` is much harder than I thought. The problem is that it is valid to say `(3,14)` so it is not safe to just detect the comma while parsing numbers and show a nice message.

I would love to show a friendly hint for this though! So please open a new issue specifically about this problem so it does not get lost!

**Comment by evancz** [comment]
I am finishing up a revamp of the parser that makes it easier to give specific messages for syntax problems, so I ended up adding a little tweak for this case:

```
-- EXTRA COMMA -------------------------------------------------------- temp.elm

I am partway through parsing a record type, but I got stuck here:

3|     { start : Position,
4|     , current : Position
       ^
I am seeing two commas in a row. This is the second one!

Just delete one of the commas and you should be all set!

Note: If you are trying to define a record type across multiple lines, I
recommend using this format:

    { name : String
    , age : Int
    , height : Float
    }

Notice that each line starts with some indentation. Usually two or four spaces.
I know this style can be jarring for people coming from C-like syntax, but folks
generally report that they are used to it (and often prefer it!) after a week or
two of using Elm.

```

Thanks again for the report! This should become available once Elm 0.19.1 is released.
